File: LMIS/LMIS.Web/Modules/InventoryManage/UpBookshelfs/UpBookshelfsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.InventoryManage.Columns;

[ColumnsScript("InventoryManage.UpBookshelfs")]
[BasedOnRow(typeof(UpBookshelfsRow), CheckNames = true)]
public class UpBookshelfsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long UpBookshelfId { get; set; }
    public string BookName { get; set; }
    public string BookshelfLocation { get; set; }
    public long Inventory { get; set; }
    public string OperateUserName { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/InventoryManage/UpbookshelfsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UpBookshelfsRow } from "./UpBookshelfsRow";

export interface UpBookshelfsColumns {
    UpBookshelfId: Column<UpBookshelfsRow>;
    BookName: Column<UpBookshelfsRow>;
    BookshelfLocation: Column<UpBookshelfsRow>;
    Inventory: Column<UpBookshelfsRow>;
    OperateUserName: Column<UpBookshelfsRow>;
    CreateTime: Column<UpBookshelfsRow>;
    UpdateTime: Column<UpBookshelfsRow>;
}

export class UpBookshelfsColumns extends ColumnsBase<UpBookshelfsRow> {
    static readonly columnsKey = 'InventoryManage.UpBookshelfs';
    static readonly Fields = fieldsProxy<UpBookshelfsColumns>();
}